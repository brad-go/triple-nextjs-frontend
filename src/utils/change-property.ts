export const changePropertyCamelToKebab = (property: string) => {
  const capital = property.match(/[A-Z]/g)

  if (capital) {
    const capitalIndex = property.indexOf(capital[0])
    const firstWord = property.slice(0, capitalIndex)
    const secondWord = capital[0]
      .toLocaleLowerCase()
      .concat(property.slice(capitalIndex + 1))

    property = `${firstWord}-${secondWord}`
  }

  return property
}
