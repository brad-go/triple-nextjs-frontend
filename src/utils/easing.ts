export function easeOutExpo(
  time: number,
  start: number,
  end: number,
  duration: number,
) {
  if (time === duration) {
    return start + end
  }

  return end * (-Math.pow(2, (-10 * time) / duration) + 1) + start
}
