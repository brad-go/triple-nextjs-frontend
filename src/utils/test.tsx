import { PropsWithChildren, ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import { theme } from 'styles/theme'

function Provider<T>({ children }: PropsWithChildren<T>) {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: Provider, ...options })

export { customRender as render }
