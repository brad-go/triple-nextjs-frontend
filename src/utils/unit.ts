export const setUnit = (value: number | string, unit = 'px') => {
  if (typeof value === 'string') {
    return value
  }

  return `${value}${unit}`
}
