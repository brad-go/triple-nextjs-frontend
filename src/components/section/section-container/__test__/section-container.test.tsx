import { screen } from '@testing-library/react'

import SectionContainer from '../section-container'

import { render } from 'utils/test'

describe('SectionContainer', () => {
  it('should render Section Container', () => {
    const { container } = render(<SectionContainer />)

    expect(container).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('should render children', () => {
    const children = <div>Example</div>
    const { container } = render(
      <SectionContainer>{children}</SectionContainer>,
    )

    expect(screen.getByText('Example')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('should have default background style', () => {
    const DEFAULT_BACKGROUND_STYLES = [
      'background-image',
      'background-repeat',
      'background-attachment',
      'background-origin',
      'background-clip',
      'background-color',
    ]

    render(<SectionContainer data-testid="container" />)

    const container = screen.getByTestId('container')
    expect(container).toHaveStyle('position: relative')
    expect(container).toHaveStyle('background-position: center center')
    expect(container).toHaveStyle('background-size: cover')
    expect(container).toHaveStyle('overflow: hidden')

    DEFAULT_BACKGROUND_STYLES.forEach((property) => {
      expect(screen.getByTestId('container')).toHaveStyle(
        `${property}: inherit`,
      )
    })
  })

  it('should have primary background style', () => {
    render(
      <SectionContainer data-testid="container" backgroundStyle="primary" />,
    )

    const container = screen.getByTestId('container')
    expect(container).toHaveStyle('background-color: rgb(250, 250, 250)')
  })

  it('should have secondary background style', () => {
    render(
      <SectionContainer data-testid="container" backgroundStyle="secondary" />,
    )

    const container = screen.getByTestId('container')
    expect(container).toHaveStyle('background-color: rgb(245, 245, 245)')
  })

  it('should have toolbox background style', () => {
    render(
      <SectionContainer data-testid="container" backgroundStyle="toolbox" />,
    )

    const container = screen.getByTestId('container')
    const URL = 'https://triple.guide/intro/images/bg-section-06.svg'

    expect(container).toHaveStyle('background-color: rgb(38, 206, 194)')
    expect(container).toHaveStyle(`background-image: url(${URL})`)
  })

  it('should have recruit background style', () => {
    render(
      <SectionContainer data-testid="container" backgroundStyle="recruit" />,
    )

    const container = screen.getByTestId('container')
    const URL = 'https://triple.guide/intro/images/group-section-8.svg'

    expect(container).toHaveStyle('background-color: rgb(54, 102, 255)')
    expect(container).toHaveStyle(`background-image: url(${URL})`)
  })

  it('should have footer background style', () => {
    render(
      <SectionContainer data-testid="container" backgroundStyle="footer" />,
    )

    const container = screen.getByTestId('container')
    expect(container).toHaveStyle('background-color: rgba(35, 51, 50, 0.98)')
  })
})
