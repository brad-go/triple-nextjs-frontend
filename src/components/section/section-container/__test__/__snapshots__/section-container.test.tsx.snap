// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionContainer should render Section Container 1`] = `
.c0 {
  position: relative;
  background-position: center center;
  background-size: cover;
  overflow: hidden;
  background-image: inherit;
  background-repeat: inherit;
  background-attachment: inherit;
  background-origin: inherit;
  background-clip: inherit;
  background-color: inherit;
}

<div>
  <section
    class="c0"
  />
</div>
`;

exports[`SectionContainer should render children 1`] = `
.c0 {
  position: relative;
  background-position: center center;
  background-size: cover;
  overflow: hidden;
  background-image: inherit;
  background-repeat: inherit;
  background-attachment: inherit;
  background-origin: inherit;
  background-clip: inherit;
  background-color: inherit;
}

<div>
  <section
    class="c0"
  >
    <div>
      Example
    </div>
  </section>
</div>
`;
